src/util/location.h

```diff
@@ -55,9 +55,18 @@ struct Location {
   }
 
   bool operator < (const Location &l) const {
-    if (filename == l.filename) { return start < l.start; }
+    if (filename == l.filename) {
+      if (start == l.start) {
+        return end < l.end;
+      }
+      return start < l.start;
+    }
     return filename < l.filename;
```

Comment:
So, to prevent nesting when comparing multiple tuple elements, people usually write this as:
```
if (a != b) return a < b;
if (c != d) return c < d;
if (e != f) return e < f;
return g < h;
```

Reply:
Done